src/Box.tsx

```diff
@@ -21,4 +21,5 @@ Box.propTypes = {
   theme: PropTypes.object,
 }
 
+export type BoxProps = React.ComponentProps<typeof Box>
 export default Box
```

Comment:
Is it worth re-considering using `default` exports as well? It is probably a breaking change if anyone was consuming directly from `@primer/components/lib/Box` or something like that, but if we're going to be making other breaking changes, it might be a fair time to consider it.

I'm not really aware of the main pros/cons of default exports, and it seems like in other code at GitHub we're inconsistent on this already, so maybe it's not even worth looking at 😅 

For example, https://github.com/github/details-dialog-element/blob/main/src/index.ts uses a default export, and web components in the dotcom codebase itself seem to be about 50/50.

Just thought I'd mention it to see if it sparks thoughts for others.

Reply:
Generally I prefer not to use default exports. Here is a good article about it: https://basarat.gitbook.io/typescript/main-1/defaultisbad

Reply:
I'm open to moving away from default exports for our components. A few thoughts:
* If we decide to remove default exports, I'd prefer to do that in a separate PR that updates all the exports of all the components at once. This would allow consumers to update all their imports at once.
* How do we ensure consistency across the codebase? Is there an existing linting rule for this?

Reply:
https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-default-export.md